tests/models/patchtst/test_modeling_patchtst.py

```diff
@@ -0,0 +1,384 @@
+        mask_type="random",
+        random_mask_ratio=0,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.prediction_length = prediction_length
+        self.context_length = context_length
+        self.patch_length = patch_length
+        self.patch_stride = patch_stride
+        self.num_input_channels = num_input_channels
+        self.num_time_features = num_time_features
+        self.is_training = is_training
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.intermediate_size = intermediate_size
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.mask_type = mask_type
+        self.random_mask_ratio = random_mask_ratio
+
+        self.seed = seed
+        self.num_targets = num_targets
+        self.distil = distil
+        self.num_patches = (max(self.context_length, self.patch_length) - self.patch_length) // self.patch_stride + 1
+
+    def get_config(self):
+        return PatchTSTConfig(
+            prediction_length=self.prediction_length,
+            patch_length=self.patch_length,
+            patch_stride=self.patch_stride,
+            num_input_channels=self.num_input_channels,
+            d_model=self.hidden_size,
+            num_hidden_layers=self.num_hidden_layers,
+            num_attention_heads=self.num_attention_heads,
+            ffn_dim=self.intermediate_size,
+            dropout=self.hidden_dropout_prob,
+            attention_dropout=self.attention_probs_dropout_prob,
+            context_length=self.context_length,
+            activation_function=self.hidden_act,
+            seed=self.seed,
+            num_targets=self.num_targets,
+            mask_type=self.mask_type,
+            random_mask_ratio=self.random_mask_ratio,
+        )
+
+    def prepare_patchtst_inputs_dict(self, config):
+        _past_length = config.context_length
+        # bs, num_input_channels, num_patch, patch_len
+
+        # [bs x seq_len x num_input_channels]
+        past_values = floats_tensor([self.batch_size, _past_length, self.num_input_channels])
+
+        future_values = floats_tensor([self.batch_size, config.prediction_length, self.num_input_channels])
+
+        inputs_dict = {
+            "past_values": past_values,
+            "future_values": future_values,
+        }
+        return inputs_dict
+
+    def prepare_config_and_inputs(self):
+        config = self.get_config()
+        inputs_dict = self.prepare_patchtst_inputs_dict(config)
+        return config, inputs_dict
+
+    def prepare_config_and_inputs_for_common(self):
+        config, inputs_dict = self.prepare_config_and_inputs()
+        return config, inputs_dict
+
+
+@require_torch
+class PatchTSTModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCase):
```

Comment:
This should have the `GenerationTesterMixin` mixin

Reply:
this model doesnt support auto-regressive generation and also doesn't have input id or tokenizer and that is why we didn't add the `GenerationTesterMixin`

Reply:
Makes sense :) Noting here so it's documented for any future PR readers: discussed offline and this means `all_generative_model_classes` can then be removed